nexus/external-api/src/lib.rs

```diff
@@ -3498,6 +3504,163 @@ pub trait NexusExternalApi {
         rqctx: RequestContext<Self::Context>,
         params: TypedBody<params::DeviceAccessTokenRequest>,
     ) -> Result<Response<Body>, HttpError>;
+
+    // Webhooks
+
+    /// List webhook event classes
+    #[endpoint {
+        method = GET,
+        path = "/v1/webhooks/event-classes",
+        tags = ["system/webhooks"],
+    }]
+    async fn webhook_event_class_list(
+        rqctx: RequestContext<Self::Context>,
+        pag_params: Query<
+            PaginationParams<EmptyScanParams, params::EventClassPage>,
+        >,
+        filter: Query<params::EventClassFilter>,
+    ) -> Result<HttpResponseOk<ResultsPage<views::EventClass>>, HttpError>;
+
+    /// Fetch details on an event class by name.
+    #[endpoint {
+        method = GET,
+        path ="/v1/webhooks/event-classes/{name}",
+        tags = ["system/webhooks"],
+    }]
+    async fn webhook_event_class_view(
+        rqctx: RequestContext<Self::Context>,
+        path_params: Path<params::EventClassSelector>,
+    ) -> Result<HttpResponseOk<views::EventClass>, HttpError>;
+
+    /// List webhook receivers.
+    #[endpoint {
+        method = GET,
+        path = "/v1/webhooks/receivers/",
+        tags = ["system/webhooks"],
+    }]
+    async fn webhook_receiver_list(
+        rqctx: RequestContext<Self::Context>,
+        query_params: Query<PaginatedByNameOrId>,
+    ) -> Result<HttpResponseOk<ResultsPage<views::WebhookReceiver>>, HttpError>;
+
+    /// Get the configuration for a webhook receiver.
+    #[endpoint {
+        method = GET,
+        path = "/v1/webhooks/receivers/{receiver}",
+        tags = ["system/webhooks"],
+    }]
+    async fn webhook_receiver_view(
+        rqctx: RequestContext<Self::Context>,
+        path_params: Path<params::WebhookReceiverSelector>,
+    ) -> Result<HttpResponseOk<views::WebhookReceiver>, HttpError>;
+
+    /// Create a new webhook receiver.
```

Comment:
Should remove "a", "new", and the full stop to match the styling of these descriptions for the docs. E.g. "Create webhook receiver"

Reply:
Also remove the periods. These become the docs sidebar text.

<img width="260" alt="image" src="https://github.com/user-attachments/assets/e2a4ee16-6e05-41e8-b9ce-092ba1801bbb" />

Here are my suggestions.

```
Create webhook receiver
Fetch webhook receiver
Delete webhook receiver
List delivery attempts to webhook receiver
Request re-delivery of webhook event
List secret IDs for webhook receiver
Add secret to webhook
```

Note `get -> fetch` for consistency with the other endpoints. To be honest I think we could stand to use get everywhere instead and save two letters, but it's where we are right now.

Reply:
I was commenting on an outdated version of the list. Will update in a minute with suggestions for the full list.

Reply:
Thanks for the suggestions, I'll reword these. I  notice that "get" and "fetch" are used pretty interchangeably in other APIs...we should probably give that a pass to make it more consistent (separately from this PR of course): 
![image](https://github.com/user-attachments/assets/c877aa39-9efa-440b-855c-721fec112379)

Reply:
https://github.com/oxidecomputer/omicron/issues/7799

Reply:
Okay, 4f8c258fb9dadee55ab59f0b7c4ea2b207ce3606 reworks the API docs as you've suggested. I also added body text to a few of the endpoints where it felt like there were additional bits that needed a more detailed explanation.